File: android/src/main/kotlin/com/mytiki/sdk/capture/receipt/capacitor/rsp/RspInitialized.kt
/*
 * Copyright (c) TIKI Inc.
 * MIT license. See LICENSE file in root directory.
 */

package com.mytiki.sdk.capture.receipt.capacitor.rsp

import org.json.JSONObject

/**
 * Represents the response data for the initialization of a specific component.
 *
 * @property isInitialized Indicates whether the component is initialized.
 */
class RspInitialized(private val isInitialized: Boolean) : Rsp {

    /**
     * Converts the response data to a JSON object.
     *
     * @return A JSON object representing the response data.
     */
    override fun toJson(): JSONObject =
        JSONObject()
            .put("isInitialized", isInitialized)
}

File: ios/Plugin/Rsp/RspInitialized.swift
/*
 * RspInitialized Struct
 * Copyright (c) TIKI Inc.
 * MIT license. See LICENSE file in root directory.
 */

import Foundation
import Capacitor

/**
 Represents a response indicating whether the ReceiptCapture plugin is initialized.

 This struct is used to convey information about the initialization status of the ReceiptCapture plugin.
 */
struct RspInitialized: Rsp {
    /// A flag indicating whether the plugin is initialized.
    private let isInitialized: Bool

    /**
     Initializes an `RspInitialized` struct.

     - Parameter isInitialized: A boolean flag indicating whether the ReceiptCapture plugin is initialized.
     */
    init(isInitialized: Bool) {
        self.isInitialized = isInitialized
    }

    /**
     Converts the `RspInitialized` struct into a dictionary suitable for use in plugin response data.

     - Returns: A dictionary containing the initialization status in a format suitable for a Capacitor plugin call result.
     */
    func toPluginCallResultData() -> Capacitor.PluginCallResultData {
        return [
            "isInitialized": isInitialized
        ]
    }
}

